include/deal.II/lac/trilinos_sparse_matrix.h

```diff
@@ -2223,10 +2225,33 @@ namespace TrilinosWrappers
         * inverse operations.
         * The <tt>vmult</tt> and <tt>Tvmult</tt> operations retain the standard
         * definitions inherited from @p op.
+        *
+        * @note This function is enabled only if the solver and preconditioner
+        * derive from the respective TrilinosWrappers base classes.
         */
         template <typename Solver, typename Preconditioner>
-        TrilinosPayload inverse_payload (Solver &, const Preconditioner &) const;
+        typename std::enable_if<
+        std::is_base_of<TrilinosWrappers::SolverBase,Solver>::value and
+        std::is_base_of<TrilinosWrappers::PreconditionBase,Preconditioner>::value,
+            TrilinosPayload>::type
+            inverse_payload (Solver &, const Preconditioner &) const;
 
+        /**
+        * Returns a payload configured for inverse operations
```

Comment:
astyle doesn't re-indent comments. At least in my browser, the asterisks do not line up.